### PR TITLE
feat: migrate components to new arch

### PR DESCRIPTION
**Pull request by WoLewicki**
PR migrating rest of the components to new arch and cleaning some code.

### CODE REVIEW

**Comment by WoLewicki** [comment]
@mfazekas I have 2 questions here:

1. Is `refresh` used in `Image` component? It is not implemented on `iOS` from what I can see.
2. There are some props on `Android` with a `TODO` in them, I am not sure what to do with them, so it would be nice to give an answer for it 🚀

File: android/src/main/java/com/rnmapbox/rnmbx/components/styles/terrain/RNMBXTerrainManager.kt [hunk 1]
```diff
@@ -14,19 +16,20 @@ class RNMBXTerrainManager : ViewGroupManager<RNMBXTerrain>() {
         return RNMBXTerrain(reactContext)
     }
 
-    @ReactProp(name = "id")
-    fun setId(layer: RNMBXTerrain, id: String?) {
-        layer.iD = id
-    }
+    // TODO: it is not present in props so should we add it?
```
Comment (mfazekas) [inline android/src/main/java/com/rnmapbox/rnmbx/components/styles/terrain/RNMBXTerrainManager.kt:19]: `id` is not needed for terrain it's a singleton

**Comment by mfazekas** [comment]
> @mfazekas I have 2 questions here:
> 
> 1. Is `refresh` used in `Image` component? It is not implemented on `iOS` from what I can see.

Yes `refresh` is needed, just did not get to implement that on iOS yet. Leave it as no-op on iOS. We need that if the child view of `Image` component has an `Image` for example that doesn't draw immediately, and we'll need to call `refresh` when the image has loaded.

> 2. There are some props on `Android` with a `TODO` in them, I am not sure what to do with them, so it would be nice to give an answer for it 🚀

**Comment by WoLewicki** [comment]
Ok I pushed the changes with `Image` module added. Could you check if it works correctly, and, if so, are we ready to merge this one too :rocket: ?

**Comment by mfazekas** [comment]
Looks good to me, v11 on iOS needs to be fixed, but that can be done later.

**Pull request merged**